hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/state/WritableStates.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2020-2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.state;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Set;
+
+/** Essentially, a map of {@link WritableKVState}s. Each state may be retrieved by key. */
+public interface WritableStates {
```

Comment:
By jove, I think you're right! I didn't think I could do it because it would conflate the `get` method -- how do you know if you want a readable state or a writable state? But if it is true that you only ever want writable KV states from a WritableStates then we can easily extend it.